tests/unit/generator/test_generator.py

```diff
@@ -125,31 +125,49 @@ def test_get_response_ignores_unwanted_transports():
             "foo/%service/transports/grpc.py.j2",
             "foo/%service/transports/__init__.py.j2",
             "foo/%service/transports/base.py.j2",
+            "foo/%service/async_client.py.j2",
+            "foo/%service/client.py.j2",
             "mollusks/squid/sample.py.j2",
         ]
 
         with mock.patch.object(jinja2.Environment, "get_template") as gt:
             gt.return_value = jinja2.Template("Service: {{ service.name }}")
+            api_schema = make_api(
+                make_proto(
+                    descriptor_pb2.FileDescriptorProto(
+                        service=[
+                            descriptor_pb2.ServiceDescriptorProto(
+                                name="SomeService"),
+                        ]
+                    ),
+                )
+            )
+
             cgr = g.get_response(
-                api_schema=make_api(
-                    make_proto(
-                        descriptor_pb2.FileDescriptorProto(
-                            service=[
-                                descriptor_pb2.ServiceDescriptorProto(
-                                    name="SomeService"),
-                            ]
-                        ),
-                    )
-                ),
+                api_schema=api_schema,
                 opts=Options.build("transport=river+car")
             )
-
-            assert len(cgr.file) == 4
+            assert len(cgr.file) == 5
             assert {i.name for i in cgr.file} == {
                 "foo/some_service/transports/river.py",
                 "foo/some_service/transports/car.py",
                 "foo/some_service/transports/__init__.py",
                 "foo/some_service/transports/base.py",
+                # Only generate async client with grpc transport
```

Comment:
nit: This comment looks as thought it applies to the next line, but is a bit confusing. Suggest moving it below the next line and tweaking it to be something like: `# no async_client.py, as that's only for grpc transport`